geomesa-native-api/src/test/java/org/locationtech/geomesa/api/GeoMesaIndexTest.java

```diff
@@ -0,0 +1,81 @@
+package org.locationtech.geomesa.api;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.gson.Gson;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+import org.junit.Assert;
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+public class GeoMesaIndexTest {
+
+    public static class DomainObject {
+        public String id;
+        public int intValue = 0;
+        public double doubleValue = 0.0;
+
+        public DomainObject(String id, int intValue, double doubleValue) {
+            this.id = id;
+            this.intValue = intValue;
+            this.doubleValue = doubleValue;
+        }
+    }
+
+    public static class DomainObjectValueSerializer implements ValueSerializer<DomainObject> {
+        public static Gson gson = new Gson();
+        @Override
+        public byte[] toBytes(DomainObject o) {
+            return gson.toJson(o).getBytes();
+        }
+
+        @Override
+        public DomainObject fromBytes(byte[] bytes) {
+            return gson.fromJson(new String(bytes), DomainObject.class);
+        }
+    }
+
+    @Test
+    public void testNativeAPI() {
+        DomainObject one = new DomainObject("1", 1, 1.0);
+        DomainObject two = new DomainObject("2", 2, 2.0);
+
+        GeoMesaIndex<DomainObject> index = AccumuloGeoMesaIndex.build("hello", "zoo1:2181", "mycloud", "myuser", "mypass", true, new DomainObjectValueSerializer());
```

Comment:
you can always create the classic Java solution of making an Opts Class or a Config class to avoid all the parameter passing in the method :)